MALLOC_PERTURB_ will make inval-equ-1.l fail, because with errors fixups aren't
run, which means the space for displacement operands remains uninitialized.
The testcase invalidly expects zeros there.  See bnc #590820 .

--- gas/testsuite/gas/i386/inval-equ-1.l.mm	2008-03-03 15:28:58.000000000 +0000
+++ gas/testsuite/gas/i386/inval-equ-1.l	2010-03-24 16:16:30.000000000 +0000
@@ -9,6 +9,6 @@ GAS LISTING .*
 [ 	]*3[ 	]+\.equ    bar1,%eax
 [ 	]*4[ 	]+\.equ    bar2,%eax
 [ 	]*5[ 	]+\.globl  bar2
-[ 	]*6[ 	]+\?\?\?\? A1000000 		mov bar1,%eax
-[ 	]*6[ 	]+00
-[ 	]*7[ 	]+\?\?\?\? 89C0     		mov bar2,%eax
+[ 	]*6[ 	]+.*[ 	]+mov bar1,%eax
+[ 	]*6[ 	]+.*
+[ 	]*7[ 	]+.*[ 	]+mov bar2,%eax
--- gas/testsuite/gas/i386/inval-equ-2.l.mm	2008-03-03 15:28:58.000000000 +0000
+++ gas/testsuite/gas/i386/inval-equ-2.l	2010-03-24 16:18:05.000000000 +0000
@@ -7,13 +7,13 @@ GAS LISTING .*
 
 [ 	]*1[ 	]+\.globl  bar1
 [ 	]*2[ 	]+\.set    bar1,\(%eax\+1\)
-[ 	]*3[ 	]+\?\?\?\? A12A0000 		mov bar1,%eax
-[ 	]*3[ 	]+00
+[ 	]*3[ 	]+.*[ 	]+mov bar1,%eax
+[ 	]*3[ 	]+.*
 [ 	]*4[ 	]+\.set    bar2,\(%eax\+1\)
-[ 	]*5[ 	]+\?\?\?\? A12A0000 		mov bar2,%eax
-[ 	]*5[ 	]+00
+[ 	]*5[ 	]+.*[ 	]+mov bar2,%eax
+[ 	]*5[ 	]+.*
 [ 	]*6[ 	]+\.globl  bar2
 [ 	]*7[ 	]+\.set    bar3,\(%eax\+1\)
-[ 	]*8[ 	]+\?\?\?\? A12A0000 		mov bar3,%eax
+[ 	]*8[ 	]+.*[ 	]+mov bar3,%eax
 \*\*\*\*  Error:can't make global register symbol `bar3'
-[ 	]*8[ 	]+00
+[ 	]*8[ 	]+.*
